### PR TITLE
Ruby Course: Use official Ruby 3.2 documentation (docs.ruby-lang.org) in place of 3rd-party and outdated sources

### LINKED ISSUE #26960: Ruby Course: Use official documentation (docs.ruby-lang.org) instead of ad-supported 3rd party (ruby-doc.org)
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

It is all too easy to confuse the 3rd-party, ad-supported website, `ruby-doc.org`, for official Ruby documentation, which it is not.

Besides this site's official-sounding name, The Odin Project's curriculum contains far more links to it (34) than to the official Ruby documentation, `docs.ruby-lang.org` (4).

This confusion is not unique to The Odin Project and has been discussed for example in [this Reddit thread](https://www.reddit.com/r/ruby/comments/14sw4px/whats_the_difference_between_rubydocorg_and/), which also gives a reproducible example of an additional issue with 3rd-party Ruby documentation:

- Certain intra- and inter-document links fail on sites like `ruby-doc.org` while functioning correctly on the official site, `docs.ruby-lang.org`.

You can reproduce the broken link test as follows:

1. Load your Ruby documentation site of choice.
2. Find and open the page, "OptionParser".
3. Search on this page for "Parameters for New Options" and see if the link works correctly.

I encountered issues like this on a daily basis when I started working through TOP's Ruby Course & mistakenly assumed the blame lay somehow with the maintainers of Ruby. My ad-blocker made it even less obvious that `ruby-doc.org` was a 3rd-party site.

My current favorite 3rd-party documentation site, `DevDocs.io` (maintained by freeCodeCamp & [listed as an additional resource](https://www.theodinproject.com/lessons/foundations-introduction-to-html-and-css#additional-resources) in one TOP Foundations lesson) -- passes this test. But multiple other 3rd-party sites including `ruby-doc.org` fail it. And I think it's a better policy to link to a language's official documentation than to any 3rd-party documentation.

 Thankfully, TOP's Ruby documentation links should be easy to fix in bulk by doing a curriculum-wide find and replace of:
`https://ruby-doc.org/*/`
with:
`https://docs.ruby-lang.org/en/3.2/`
as the two sites have the same URL endings. (At least, this worked on the links I tested so far, and it would be easy to test the others and make adjustments if needed.)

I am happy to make a PR for this change if approved.

### Path

Ruby / Rails

### Lesson Url

https://www.theodinproject.com/lessons/ruby-basic-data-types

### (Optional) Discord Name

macauleydev

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by macauleydev**
## Because

- Most of TOP's links to Ruby documentation point to 3rd-party sites which currently have ads, broken internal links, and/or outdated and missing materials.
- There is confusion about the provenance of official Ruby documentation. A tentative partial explanation is that the 3rd-party site `ruby-doc.org` (now ad-supported and with many broken and outdated links) was many years ago superior to the official Ruby site, `ruby-lang.org`. The former is still referenced widely in the Ruby community (including [until recently](https://launchschool.com/books/ruby/read/preparations#readingdocumentation) by Launch School, one of TOP's main sources in the early lessons of the Ruby Course).
- Currently, the official Ruby documentation site ([docs.ruby-lang.org](https://docs.ruby-lang.org/en/3.2/)) appears better maintained and with fewer issues (such as broken links) than any of the 3rd party Ruby documentation sites linked in TOP's curriculum: ruby-doc.org, rubydoc.info, rubyapi.org.

## This PR

In lessons throughout the Ruby Course:
- Updates the handful of existing `docs.ruby-lang.org` links from Ruby 3.0 to Ruby 3.2.
- Updates the 30+ `ruby-doc.org` links (mostly Ruby 3.1) to `docs.ruby-lang.org` and Ruby 3.2.
- Updates the 30+ `rubydoc.info` links (mostly Ruby 3.0) to `docs.ruby-lang.org` and Ruby 3.2.
- Updates the handful of `rubyapi.org` links (Ruby 3.1) to `docs.ruby-lang.org` and Ruby 3.2.
- Updates the instructions on navigating Ruby documentation (in the [Arrays lesson](https://www.theodinproject.com/lessons/ruby-arrays)) to match the structure of `docs.ruby-lang.org` instead of `ruby-doc.org`.

## Issue

Closes #26960

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby/basic_ruby/arrays.md [hunk 1]
```diff
@@ -139,14 +139,14 @@ Here is a brief look at some other common array methods you might run into:
...
 </div>
 
 ### Additional resources
 This section contains helpful links to other content. It isn't required, so consider it supplemental.
 
-* Look over the latest [Ruby API documentation](http://ruby-doc.org/) on arrays by clicking on the "Current" tab in the top left corner, scrolling down to the "Class and Module Index" section on the left sidebar, and clicking on "Array". There, you'll find the most up-to-date documentation on the various methods available to the Array object along with explanations and examples.
+* Look over the latest [Ruby API documentation](https://docs.ruby-lang.org/) on arrays by choosing a language and Ruby version (e.g. English, Ruby 3.2), then scrolling down to the "Class and Module Index" section on the left sidebar and clicking on "Array". There, you'll find the most up-to-date documentation on the various methods available to the Array object along with explanations and examples.
```
Comment (rlmoser99) [inline ruby/basic_ruby/arrays.md:149]: What do you think about just linking to https://docs.ruby-lang.org/en/3.2/Array.html here?

**Review by rlmoser99 — COMMENTED** [review]
Thanks for taking the time to update these links. I have left one question for us to discuss.

File: ruby/basic_ruby/arrays.md (hunk 1 above)
Comment (macauleydev) [reply ruby/basic_ruby/arrays.md:149]: I think it's a great idea to at least eliminate this second explanation on line 149, replacing it with a simple link.

One could also argue for a similar change to the first explanation on line 111. I think I would agree with that, but I'm not attached.

Comment (rlmoser99) [reply ruby/basic_ruby/arrays.md:149]: Oh, yes! Making a change to line 111 would be useful too. 👍

**Review by rlmoser99 — APPROVED** [review]
Thank you so much for taking the time to make this PR and discuss the additional changes.

**Comment by macauleydev** [comment]
Of course! Thanks for the feedback and for merging so quickly.

**Pull request merged**